setup.cfg

```diff
@@ -55,7 +55,85 @@ test =
     hypothesis>=5.5.3
     pytest>=6.0
     pytest-xdist>=1.31
-
+# optional extras for recommended dependencies
```

Comment:
Could these be specified in `pyproject.toml` instead?

Reply:
Yes. `pyproject.toml [project.optional-dependencies]` and and `setup.cfg [options.extras_require]` should be equivalent.  The ecosystem has been changing rapidly over the past few years.

For reference: [PEP-621](https://peps.python.org/pep-0621/#dependencies-optional-dependencies)

If moving this to pyproject.toml, then we should also move over the existing `test = {stuff}` extra.  For cleanliness, I'd suggest merging this PR as-is, and then porting all of this over to `toml` in another PR.

While Python community is moving towards toml as ground-truth for project definitions, I think pandas core  team should discuss before making the change to dependency declaration in `pyproject.toml`

Reply:
Yeah agreed. We've been trying to prefer `pyproject.toml` over `setup.cfg` but agreed that can be done in another PR